packages/@aws-cdk/cloudformation-diff/test/diff-template.test.ts

```diff
@@ -1231,4 +1231,22 @@ describe('changeset', () => {
     expect(differences.resources.differenceCount).toBe(1);
     expect(differences.resources.get('BucketResource').changeImpact === ResourceImpact.WILL_IMPORT);
   });
+
+  test('does not do changeset checks for new stack', async () => {
```

Comment:
I might be missing something but I'm not sure how this tests that we do not do a changeset for new stacks. Could you clarify that?

Reply:
Was confused on this too - my logic was that calling `fullDiff` with the current stack being empty (new) should mean that there should be only 1 difference count, and the change impact should be `ResourceImpact.WILL_CREATE`; I'm not sure how else to test that we don't do changeset for new stacks